Products/ZenUI3/browser/resources/js/zenoss/EventPanels.js

```diff
@@ -1756,12 +1756,15 @@
             }
         },
         refresh: function(){
-            this.directFn({uid:this.uid, keys:['events']}, function(result){
-                if (Zenoss.env.contextUid && Zenoss.env.contextUid !== this.uid) {
-                    return;
-                }
-                this.updateRainbow(result.data.events);
-            }, this);
+            // we always need uid - so no uid no refresh;
```

Comment:
What does this have to do with add device?

Reply:
While reproducing this issue i notice that when there is nothing selected or selected main root node and "Rainbow" component refreshes we get server error that some function needs two arguments, and i decide that this could be related and easy fixed with this issue.

Reply:
@jplouis  please finish review of this ticket.

Reply:
I'm not seeing this as an issue on alphacorp or betacorp sites. Is this still an issue?

Reply:
it could be an issue on first page load when nothing is selected in device tree.

Reply:
this was changed in scope of ticket ZEN-30131